addons/website_slides/models/slide_slide.py

```diff
@@ -533,6 +534,16 @@ def _compute_website_url(self):
                 base_url = slide.channel_id.get_base_url()
                 slide.website_url = '%s/slides/slide/%s' % (base_url, slug(slide))
 
+    @api.depends('channel_id.website_id.domain', 'website_url')
+    def _compute_website_share_url(self):
+        for slide in self:
+            if slide.id:  # ensure we can build the URL
+                if slide.is_preview:
+                    slide.website_share_url = slide.website_url
```

Comment:
Actually couldn't it be managed directly by the /share link ? That way if you change the is_preview flag the links are still valid and controller handles it gracefully.